src/mono/mono/metadata/marshal.c

```diff
@@ -6580,3 +6580,288 @@ mono_wrapper_caches_free (MonoWrapperCaches *cache)
 	free_hash (cache->thunk_invoke_cache);
 	free_hash (cache->unsafe_accessor_cache);
 }
+
+#ifdef MONO_ARCH_HAVE_SWIFTCALL
+typedef enum {
```

Comment:
The ifdefs is not needed, if its not called, it will be linked out. Also, the ifdef is defined in mini-<arch>.h in mini/ so its defined here.